src/utils/Grid.h

```diff
@@ -88,8 +84,40 @@ template<typename T> class Grid final
        */
       void resize(const geometry::Size& size, const T& defaultValue = T())
       {
-         m_width = size.width;
-         m_grid.resize(size.width * size.height, defaultValue);
+         m_size = size;
+         m_grid.resize(m_size.getArea(), defaultValue);
+      }
+   
+      /**
+       * Sets all grid cells in a given rectangle to a specific value.
+       *
+       * @param rect The Rectangle representing the subregion to modify in the Grid.
+       * @param value The value to fill the subregion with.
+       */
+      void fillRect(const geometry::Rectangle& rect, const T& value)
+      {
+         if(!rect.isValid())
+         {
+            return;
+         }
+
+         for(int y = rect.top; y < rect.bottom; ++y)
+         {
+            if(y < 0 || y >= m_size.height)
+            {
+               continue;
+            }
+
+            for(int x = rect.left; x < rect.right; ++x)
+            {
+               if(x < 0 || x >= m_size.width)
```

Comment:
Decided against silent bounds checking. We should assert here when we have a decent assert library.